sktime/annotation/e_agglo.py

```diff
@@ -0,0 +1,427 @@
+# -*- coding: utf-8 -*-
+"""E-agglo."""
+
+import warnings
+
+import numpy as np
+import pandas as pd
+
+from sktime.transformations.base import BaseTransformer
+
+__author__ = ["KatieBuc"]
+__all__ = ["EAGGLO"]
+
+
+class EAGGLO(BaseTransformer):
+    """
+    Docstring.
+
+    Parameters
+    ----------
+    ...
+
+    Attributes
+    ----------
+    ...
+
+    References
+    ----------
+    .. [1] ...
+
+    Examples
+    --------
+    ...
+    """
+
+    _tags = {
+        "fit_is_empty": False,
+    }
+
+    def __init__(
+        self,
+        member=None,
+        alpha=1.0,
+        penalty=None,
+    ):
+        self.member = member
+        self.alpha = alpha
+        self.penalty = penalty
+        super(EAGGLO, self).__init__()
+
+    def _process_data(self, X):
+        """Docstring."""
+        member_ = (
+            self.member if self.member is not None else np.array(range(X.shape[0]))
+        )
+
+        u = np.sort(np.unique(member_))  # unique array of cluster labels
+        N_ = len(u)  # number of clusters
```

Comment:
how about renaming `N_` to `n_cluster` the you don't need the comment to explain it